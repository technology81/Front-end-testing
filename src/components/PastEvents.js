import React, {useEffect, useState} from 'react';
import {Grid, Paper, Box, Card, CardContent, Button, Typography, makeStyles} from '@material-ui/core';
import AccountCircleIcon from '@material-ui/icons/AccountCircle';
//import logo from './logo.jpeg';
//import wkndevnt1 from './HH-pic.jpg';

import moment from 'moment';
import axios from 'axios';
import {useHistory } from 'react-router-dom';
import Homebar from "./Homebar";
import Footer from './Footer';

import Carousel from 'react-bootstrap/Carousel';
import 'bootstrap/dist/css/bootstrap.css';

const useStyles = makeStyles((theme) => ({
    root:{
        display: 'flex',
        flexWrap: 'wrap',
        '& > *': {
            margin: theme.spacing(1)
        },
    },
    grid:{
        display: 'flex',
        flexWrap: 'wrap',
    },
    card:{
        maxWidth: 175,
        display: 'flex',
        flexWrap: 'wrap',

    },
    button:{
        color:"primary",
        '&:hover':{
            backgroundColor:"#2471A3",
        },
        marginTop:"8px"
    },
    carouselCaption:{
        '&:hover':{
            backgroundColor:"#F5F5F5",
            backgroundOpacity:0.05,
        }
    }
}));



const PastEvents = (props) => {
    const gridStyle= {margin:'3px 3px', padding:'5px 5px'}
    const imgStyle = {height:'420px', width:'600px'}
    const headStyle = {margin:'0', fontFamily:'serif', color:'blue'}
    const btnStyle = {margin:'8px 0'}
    const logoStyle = {height:98, width:128}
    //const imgStyle = {height:'100px', width:'180px'}
    //const stepperStyle = {}
    let history = useHistory();


    /*const handleChange = () => {
        props.history.push('/home)')
    }*/

    const [pevent, setPevent] = useState([])
    const event = "Past event"
    const Home = () => {
        history.push('/apphome');
    };
    useEffect(() => {
        axios.get('http://localhost:8081/account/events/getEventsList/false/Weekend event')
        //('http://localhost:8081/account/events/getEvents/'.concat('/isFutureEvent').concat('past'))
        //(`http://localhost:8081/account/events/getEventsList/isFutureEvent${past}/eventTypes${event}`)
        .then(response => {
            console.log(response)
            console.log(response.data[0].event_id)
            setPevent(response.data)
        })
        .catch(err=>{
            console.log(err)

        })
    },[event])

    //Carousel

    const classes = useStyles();
    return(
        <Box>
        

        <Box align="center">
        <center>
            <Typography variant='h4' color="textSecondary" align="center">Past Events</Typography>
            <p style={{ height: "30px", width: "1200px", padding: "5px", borderRadius: "5px", color: "#2874a6" }}>
                By Elizabeth Andrew  -  <strong><i> Volunteers do not necessarily have the time; they have the heart. </i></strong>
            </p>
        </center>
        </Box>

      <Grid container className={classes.root} spacing={3}>
      <Grid item xs={6}>
            <Paper style={{height:'460px', borderRadius:'10px', width:"615px"}} variant="outlined" >
                <Carousel interval={10000} position="absolute" variant="dark" style={{marginLeft:'5px', marginRight:'5px'}} >
                  {pevent.map((post) => (
                            <Carousel.Item>
                                <img
                                    className="d-block w-100"
                                    src={`data:image/png;base64,${post.image}`}
                                    style={imgStyle}
                                    alt="HH-Event"
                                />
                                <Carousel.Caption >
                                    <h4 style={{color:"black"}} >
                                        {post.name}
                                    </h4>
                                    <p align="left" style={{color:"black"}} >
                                        <b>Venue</b>: {post.venue}
                                        <br />
                                        <b>Date</b>: {moment(post.start_time).format('MMMM Do YYYY')}
                                        <br />
                                        <b>About</b>: {post.description}
                                    </p>
                                </Carousel.Caption>
                            </Carousel.Item>
                  ))}
                </Carousel>
            </Paper>
            </Grid>
            <Grid item xs={5}>
            <Paper style={{height:'460px', borderRadius:'10px', width:"580px"}} variant="outlined" >
                <Typography align="center" variant='h5' style={{color:"textSecondary", fontFamily:"Gabriola",marginTop:"20px"}} >
                    <b>Our Volunteers' Thoughts on Helping Hands</b>
                </Typography>

                <div className={classes.root} >
                    <Grid container spacing={2}>
                        <Grid item xs >
                            <Card className={classes.card} style={{spacing:2,backgroundColor:"#4DD0E1"}} >
                                <CardContent>
                                    <Typography component='p' >
                                        <i>"Really enjoy volunteering for Helping Hands. when a little of your time and the simplest of acts can make someone's day better, it is so rewarding"</i>
                                        <br/>
                                        - Geethanjali Madhu
                                    </Typography>
                                    <center><AccountCircleIcon /></center>
                                </CardContent>
                            </Card>
                        </Grid>

                        <Grid item xs >
                            <Card className={classes.card} style={{spacing:2,backgroundColor:"#4DD0E1"}} >
                                <CardContent>
                                    <Typography component='p' >
                                       <i> "I got to experience so many elements of volunteering since joining Helping Hands, it was a genuinely enriching experience. Even besides that, I was unaware just how much fun I would have!"</i>
                                         <br/>
                                         - Kausalya R
                                    </Typography>
                                    <center><AccountCircleIcon /></center>
                                </CardContent>
                            </Card>
                        </Grid>

                        <Grid item xs >
                            <Card className={classes.card} style={{spacing:2,backgroundColor:"#4DD0E1"}} >
                                <CardContent>
                                    <Typography component='p' >
                                        <i>"If you are looking for an authentic volunteering experience among people who work passinately for the causes they believe in, then this is the place to be"</i>
                                         <br/>
                                         - Shreya Patil
                                    </Typography>
                                    <center><AccountCircleIcon /></center>
                                </CardContent>
                            </Card>
                        </Grid>


                    </Grid>
                </div>

            </Paper>
            </Grid>
                  </Grid>
    <Footer />
    </Box>
    )
}

export default PastEvents;