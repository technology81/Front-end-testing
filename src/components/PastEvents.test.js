import React from 'react';
import renderer from 'react-test-renderer';
import PastEvents from './PastEvents';

it('PastEvents', () => {
    const tree=renderer.create(<PastEvents />).toJSON();
    expect(tree).toMatchSnapshot();
})