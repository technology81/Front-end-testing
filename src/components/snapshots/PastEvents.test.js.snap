// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PastEvents 1`] = `
<div
  className="MuiBox-root MuiBox-root-6"
>
  <div
    align="center"
    className="MuiBox-root MuiBox-root-7"
  >
    <center>
      <h4
        className="MuiTypography-root MuiTypography-h4 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
      >
        Past Events
      </h4>
      <p
        style={
          Object {
            "borderRadius": "5px",
            "color": "#2874a6",
            "height": "30px",
            "padding": "5px",
            "width": "1200px",
          }
        }
      >
        By Elizabeth Andrew  -  
        <strong>
          <i>
             Volunteers do not necessarily have the time; they have the heart. 
          </i>
        </strong>
      </p>
    </center>
  </div>
  <div
    className="MuiGrid-root makeStyles-root-1 MuiGrid-container MuiGrid-spacing-xs-3"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
    >
      <div
        className="MuiPaper-root MuiPaper-outlined MuiPaper-rounded"
        style={
          Object {
            "borderRadius": "10px",
            "height": "460px",
            "width": "615px",
          }
        }
      >
        <div
          className="carousel slide"
          onKeyDown={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          position="absolute"
          style={
            Object {
              "marginLeft": "5px",
              "marginRight": "5px",
            }
          }
          variant="dark"
        >
          <ol
            className="carousel-indicators"
          />
          <div
            className="carousel-inner"
          />
          <a
            className="carousel-control-prev"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
          >
            <span
              aria-hidden="true"
              className="carousel-control-prev-icon"
            />
            <span
              className="sr-only"
            >
              Previous
            </span>
          </a>
          <a
            className="carousel-control-next"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
          >
            <span
              aria-hidden="true"
              className="carousel-control-next-icon"
            />
            <span
              className="sr-only"
            >
              Next
            </span>
          </a>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5"
    >
      <div
        className="MuiPaper-root MuiPaper-outlined MuiPaper-rounded"
        style={
          Object {
            "borderRadius": "10px",
            "height": "460px",
            "width": "580px",
          }
        }
      >
        <h5
          className="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter"
          style={
            Object {
              "color": "textSecondary",
              "fontFamily": "Gabriola",
              "marginTop": "20px",
            }
          }
        >
          <b>
            Our Volunteers' Thoughts on Helping Hands
          </b>
        </h5>
        <div
          className="makeStyles-root-1"
        >
          <div
            className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
          >
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
            >
              <div
                className="MuiPaper-root MuiCard-root makeStyles-card-3 MuiPaper-elevation1 MuiPaper-rounded"
                style={
                  Object {
                    "backgroundColor": "#4DD0E1",
                    "spacing": 2,
                  }
                }
              >
                <div
                  className="MuiCardContent-root"
                >
                  <p
                    className="MuiTypography-root MuiTypography-body1"
                  >
                    <i>
                      "Really enjoy volunteering for Helping Hands. when a little of your time and the simplest of acts can make someone's day better, it is so rewarding"
                    </i>
                    <br />
                    - Geethanjali Madhu
                  </p>
                  <center>
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
                      />
                    </svg>
                  </center>
                </div>
              </div>
            </div>
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
            >
              <div
                className="MuiPaper-root MuiCard-root makeStyles-card-3 MuiPaper-elevation1 MuiPaper-rounded"
                style={
                  Object {
                    "backgroundColor": "#4DD0E1",
                    "spacing": 2,
                  }
                }
              >
                <div
                  className="MuiCardContent-root"
                >
                  <p
                    className="MuiTypography-root MuiTypography-body1"
                  >
                    <i>
                       "I got to experience so many elements of volunteering since joining Helping Hands, it was a genuinely enriching experience. Even besides that, I was unaware just how much fun I would have!"
                    </i>
                    <br />
                    - Kausalya R
                  </p>
                  <center>
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
                      />
                    </svg>
                  </center>
                </div>
              </div>
            </div>
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
            >
              <div
                className="MuiPaper-root MuiCard-root makeStyles-card-3 MuiPaper-elevation1 MuiPaper-rounded"
                style={
                  Object {
                    "backgroundColor": "#4DD0E1",
                    "spacing": 2,
                  }
                }
              >
                <div
                  className="MuiCardContent-root"
                >
                  <p
                    className="MuiTypography-root MuiTypography-body1"
                  >
                    <i>
                      "If you are looking for an authentic volunteering experience among people who work passinately for the causes they believe in, then this is the place to be"
                    </i>
                    <br />
                    - Shreya Patil
                  </p>
                  <center>
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
                      />
                    </svg>
                  </center>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiBox-root MuiBox-root-11"
  >
    <div
      className="makeStyles-bot-10"
    >
      <div
        className="MuiBottomNavigation-root makeStyles-bnav-9"
      >
        <button
          className="MuiButtonBase-root MuiBottomNavigationAction-root MuiBottomNavigationAction-iconOnly"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiBottomNavigationAction-wrapper"
          >
            <img
              src="logo-FB.ico"
              style={
                Object {
                  "height": 25,
                  "width": 25,
                }
              }
            />
            <span
              className="MuiBottomNavigationAction-label MuiBottomNavigationAction-iconOnly"
            >
              Facebook
            </span>
          </span>
        </button>
        <button
          className="MuiButtonBase-root MuiBottomNavigationAction-root MuiBottomNavigationAction-iconOnly"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiBottomNavigationAction-wrapper"
          >
            <img
              src="logo-IG.ico"
              style={
                Object {
                  "height": 25,
                  "width": 25,
                }
              }
            />
            <span
              className="MuiBottomNavigationAction-label MuiBottomNavigationAction-iconOnly"
            >
              Instagram
            </span>
          </span>
        </button>
        <button
          className="MuiButtonBase-root MuiBottomNavigationAction-root MuiBottomNavigationAction-iconOnly"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiBottomNavigationAction-wrapper"
          >
            <img
              src="logo-YT.ico"
              style={
                Object {
                  "height": 25,
                  "width": 25,
                }
              }
            />
            <span
              className="MuiBottomNavigationAction-label MuiBottomNavigationAction-iconOnly"
            >
              Youtube
            </span>
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
